added ShowSnackBar method

Courtneytotalmobile opened the pull request:
Draft PR - added ShowSnackBar method, which accepts a message, action btn text and action callback
<img width="1179" height="2556" alt="Simulator Screenshot - iPhone 16 - 2025-12-18 at 09 03 25" src="https://github.com/user-attachments/assets/511d34e2-081a-4fb0-8a0b-06c0e7099b3b" />

Courtneytotalmobile committed:
added ShowSnackBar method

robert-sayers commented:
Comments regarding the screenshot: 

the image is being clipped 
What is the circle on the left size - theres no padding between them and the text ?

Courtneytotalmobile committed:
first approved design

Courtneytotalmobile committed:
2nd design

robert-sayers commented on BTProgressHUD/ProgressHUD.cs:
```diff
@@ -158,6 +158,51 @@ public void ShowToast(string status, MaskType maskType = MaskType.None, ToastPos
+            double timeoutMs = 3000)
+        {
+            
+            const int MaxLength = 70;
```
why are we suddenly introducing limits?

Courtneytotalmobile replied:
Removed limit and the UI is now handling the length of string

robert-sayers commented on BTProgressHUD/ProgressHUD.cs:
```diff
@@ -158,6 +158,51 @@ public void ShowToast(string status, MaskType maskType = MaskType.None, ToastPos
+                    : "...";
+            }
+
+            if (string.IsNullOrEmpty(actionText))
```
we should be able to drop the if and else and just pass the variables through, if null its the same as using defaults.

robert-sayers commented on BTProgressHUD/ProgressHUD.cs:
```diff
@@ -1035,6 +1085,140 @@ void SetProgressTimer(NSTimer newtimer)
+
+            // --- Apply Dark Mode Appearance ---
+            HudView.BackgroundColor = UIColor.FromRGB(39, 39, 41);
+            StringLabel.TextColor = UIColor.White;
```
should we not be using the exposed colors? how do we control these for a different theme

robert-sayers commented on BTProgressHUD/ProgressHUD.cs:
```diff
@@ -1035,6 +1085,140 @@ void SetProgressTimer(NSTimer newtimer)
+            
+            // Create separator view
+            UIView separator = new UIView(new CGRect(barX, barY, barWidth, barHeight));
+            separator.BackgroundColor = UIColor.White;
```
hard coding the color? we should be exposing or using thee existing color values

robert-sayers commented on BTProgressHUD/ProgressHUD.cs:
```diff
@@ -1035,6 +1085,140 @@ void SetProgressTimer(NSTimer newtimer)
+            separator.Tag = SeparatorTag;
+            
+            // Add to HUD
+            HudView.AddSubview(separator);
```
im not so sure on the separator can we see a sample without it

Courtneytotalmobile replied:
<img width="1206" height="2622" alt="Simulator Screenshot - iPhone 16 Pro - 2026-05-13 at 17 10 00" src="https://github.com/user-attachments/assets/4e808279-58f6-4b38-83f0-03f745e645fc" />
<img width="1206" height="2622" alt="Simulator Screenshot - iPhone 16 Pro - 2026-05-13 at 17 12 31" src="https://github.com/user-attachments/assets/88df8971-89ca-4082-af7b-8254924ba3b6" />

Screenshots with and without the separator have been uploaded, let me know which one is preferred

Courtneytotalmobile replied:
after discussion in team - I have pushed up agreed design changes = Removed the seperator and action button is  made bolder
<img width="1206" height="2622" alt="Simulator Screenshot - iPhone 16 Pro - 2026-05-20 at 15 22 03" src="https://github.com/user-attachments/assets/7d57c747-d0ee-46ee-82de-2a41e50e0a12" />

Courtneytotalmobile committed:
adjustments to support full width toast and snackbar

Courtneytotalmobile commented:
> Comments regarding the screenshot:
> 
> the image is being clipped What is the circle on the left size - theres no padding between them and the text ?



Added padding and removed translucency

Courtneytotalmobile committed:
Merge branch 'master' into feature/added-snackbar-with-action-btn
# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

Courtneytotalmobile committed:
addressing pr comments

Courtneytotalmobile committed:
seperator now using theme color

Courtneytotalmobile committed:
removed maxlength for snackbar message

robert-sayers commented:
Can we take a look at how its running on the ios 26s with the whole rotation issues we have

Courtneytotalmobile commented:
> Can we take a look at how its running on the ios 26s with the whole rotation issues we have

Checked on iPhone or iPad, (iOS 26) both showing expected behaviour.

iPhone: When rotated, whilst toast/snackbar is displayed - the toast/snackbar maintains position on screen along with the background UI 

iPad: When rotated, whilst toast/snackbar is displayed - the toast/snackbar rotates along with the background UI. In both landscape and portrait toast/snackbar is positioned well, with correct spacing

Courtneytotalmobile committed:
adding iPad specific margining

Courtneytotalmobile committed:
removal of divider and boldening of action title

robert-sayers commented on BTProgressHUD/ProgressHUD.cs:
```diff
@@ -695,7 +719,7 @@ UIButton CancelHudButton
                     _cancelHud.SetTitleColor(HudForegroundColor, UIControlState.Normal);
                     _cancelHud.UserInteractionEnabled = true;
-                    _cancelHud.TitleLabel.Font = HudFont;
+                    _cancelHud.TitleLabel.Font = UIFont.SystemFontOfSize(16.0f, UIFontWeight.Heavy);
```
should be configurable as a parameter/property

Courtneytotalmobile replied:
This is now configurable as a parameter

robert-sayers commented on BTProgressHUD/ProgressHUD.cs:
```diff
@@ -1035,6 +1059,106 @@ void SetProgressTimer(NSTimer newtimer)
+            // This part ensures the HUD itself is always wide
+            bool isIPad = UIDevice.CurrentDevice.UserInterfaceIdiom == UIUserInterfaceIdiom.Pad;
+            nfloat screenWidth = UIApplication.SharedApplication.KeyWindow.Bounds.Width;
+            nfloat desiredHorizontalMargin = isIPad ? 48f : 8f; 
```
can this not be calculated as a percentage of the screen width?

Courtneytotalmobile replied:
It is now been calculated as a percentage of screen width along with min and max values

Courtneytotalmobile committed:
addressing pr comments

jenniferl11 commented on BTProgressHUD/ProgressHUD.cs:
```diff
@@ -695,7 +727,7 @@ UIButton CancelHudButton
                     _cancelHud.SetTitleColor(HudForegroundColor, UIControlState.Normal);
                     _cancelHud.UserInteractionEnabled = true;
-                    _cancelHud.TitleLabel.Font = HudFont;
+                    // _cancelHud.TitleLabel.Font = UIFont.SystemFontOfSize(16.0f, UIFontWeight.Heavy);
```
is this needed?

Courtneytotalmobile replied:
removed

jenniferl11 commented on BTProgressHUD/ProgressHUD.cs:
```diff
@@ -1035,6 +1067,107 @@ void SetProgressTimer(NSTimer newtimer)
+                existingSeparator.RemoveFromSuperview();
+            }
+
+            if (!isSnackBar)
```
should the return above the Remove existing separator?

Courtneytotalmobile replied:
If we return before the separator removal block, we risk leaving a orphaned separator line on the screen if the HUD transitions dynamically from a SnackBar layout back to a standard loading layout. Keeping the cleanup step above the return ensures the UI resets correctly.

I think the goal for this is to remove the sererator from view unconditionally (at all times)

Courtneytotalmobile committed:
removed unneeded comment
